textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/queryrewriter/FuzzyTokenizer.java

```diff
@@ -43,36 +44,106 @@ public FuzzyTokenizer(String phrase) {
     }
 
     /**
+     * When called returns a single-member list containing most-likely tokenization of phrase
+     * Splits phrase into terms by whitespace delimiter
+     * For each term, rewrites(tokenizes) term to detect likely missing spaces
+     * Concatenates all rewritten terms with singlespace delimiter
+     *
+     * For example, if input phrase is "newyork city", it would first split it into "newyork" and "city"
+     * rewriteTerm on "newyork" should return "new york"
+     * rewriteTerm on "city" should return "city"
+     * The two rewritten strings are then concatenated in order by a singlespace to give "new york city"
+     * The result returned as a single-member list is {"new york city"}
+     *
+     * @return
+     */
+    public List<String> getFuzzyTokens() {
+        String[] terms = phrase.split("\\s+");
+        String rewrittenPhrase = "";
+        for(String term : terms)
+            rewrittenPhrase += " " + rewriteTerm(term);
+        rewrittenPhrase = rewrittenPhrase.substring(1);
```

Comment:
I think this statement is getting rid of the first space.  If so, can we rewrite the code without this extra step?